File: go-version/leetcode/LC1299/replace_elements.go
package lc1299

func replaceElements(arr []int) []int {
	n := len(arr)
	ans := make([]int, n)
	mx := -1
	for i := n - 1; i >= 0; i-- {
		ans[i] = mx
		mx = max(mx, arr[i])
	}
	return ans
}

File: python-version/LC1299/replace_elements.py
class Solution:
    def replaceElements(self, arr: list[int]) -> list[int]:
        n = len(arr)
        ans = [0] * n
        mx = -1
        for i in range(n-1, -1, -1):
            ans[i] = mx
            mx = max(mx, arr[i])
        return ans